crane/template.go

```diff
@@ -0,0 +1,131 @@
+package crane
+
+import (
+	"fmt"
+	"github.com/michaelsauter/crane/print"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+type TemplateInfo struct {
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+type TemplateInfoForContainer struct {
+	CurrentContainer   ContainerInfo
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+
+// Sub-interface of Container
+type ContainerInfo interface {
+	Name() string
+	Dockerfile() string
+	Image() string
+	Id() string
+	Dependencies() *Dependencies
```

Comment:
I see your point with this as well. We could remove `Dependencies` from `ContainerInfo` and check them directly from `DependencyMap`.

For the other hand, it's true that making `RunParams()` accessible could be interesting. But then `RmParams()` and `StartParams()` as well?

In fact, I was thinking to give all the information of `config` struct, but interface `Config` limit us now, so I would need to add some methods. I mean to have something like this:
```
type TemplateInfo struct {
  ContainerMap   map[string]ContainerInfo
  DependencyMap  DependencyGraph
  Target         []string
  Groups         map[string][]string
}

func (ti TemplateInfo) TargetedContainers() []ContainerInfo
func (ti TemplateInfo) Dependencies(c ContainerInfo) *Dependencies
func (ti TemplateInfo) ContainersForGroup(g string) []ContainerInfo
func (ti TemplateInfo) IsTarget(c ContainerInfo) bool
func (ti TemplateInfo) BelongsToGroup(g string) bool
// you can think more...
```